[2025-07-25T19:18:18.361+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-25T19:18:18.479+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: sample_ETL.load_order_and_delivery_data manual__2025-07-25T19:17:57.744600+00:00 [queued]>
[2025-07-25T19:18:18.497+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: sample_ETL.load_order_and_delivery_data manual__2025-07-25T19:17:57.744600+00:00 [queued]>
[2025-07-25T19:18:18.499+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-07-25T19:18:18.553+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): load_order_and_delivery_data> on 2025-07-25 19:17:57.744600+00:00
[2025-07-25T19:18:18.571+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=210) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-25T19:18:18.574+0000] {standard_task_runner.py:63} INFO - Started process 211 to run task
[2025-07-25T19:18:18.571+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'sample_ETL', 'load_order_and_delivery_data', 'manual__2025-07-25T19:17:57.744600+00:00', '--job-id', '40', '--raw', '--subdir', 'DAGS_FOLDER/sample_ETL.py', '--cfg-path', '/tmp/tmpwd_f9b4y']
[2025-07-25T19:18:18.575+0000] {standard_task_runner.py:91} INFO - Job 40: Subtask load_order_and_delivery_data
[2025-07-25T19:18:18.723+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/settings.py:195 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2025-07-25T19:18:18.927+0000] {task_command.py:426} INFO - Running <TaskInstance: sample_ETL.load_order_and_delivery_data manual__2025-07-25T19:17:57.744600+00:00 [running]> on host f1e4a4fa89e2
[2025-07-25T19:18:20.219+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='sample_ETL' AIRFLOW_CTX_TASK_ID='load_order_and_delivery_data' AIRFLOW_CTX_EXECUTION_DATE='2025-07-25T19:17:57.744600+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-25T19:17:57.744600+00:00'
[2025-07-25T19:18:20.224+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-25T19:18:20.280+0000] {logging_mixin.py:188} INFO - ðŸš€ Starting load_order_and_delivery_data
[2025-07-25T19:18:24.353+0000] {conn.py:1516} WARNING - DNS lookup failed for kafka:9092, exception was [Errno -5] No address associated with hostname. Is your advertised.listeners (called advertised.host.name before Kafka 9) correct and resolvable?
[2025-07-25T19:18:24.354+0000] {conn.py:323} ERROR - <BrokerConnection client_id=kafka-python-producer-1, node_id=bootstrap-0 host=kafka:9092 <connecting> [unspecified None]>: DNS lookup failed for kafka:9092 (0)
[2025-07-25T19:18:24.355+0000] {conn.py:945} ERROR - <BrokerConnection client_id=kafka-python-producer-1, node_id=bootstrap-0 host=kafka:9092 <connecting> [unspecified None]>: Closing connection. KafkaConnectionError: DNS failure
[2025-07-25T19:18:24.356+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-25T19:18:24.358+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/script2.py", line 67, in load_order_and_delivery_data
    producer = KafkaProducer(
               ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/kafka/producer/kafka.py", line 481, in __init__
    client = self.config['kafka_client'](
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/kafka/client_async.py", line 262, in __init__
    self.config['api_version'] = self.check_version()
                                 ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/kafka/client_async.py", line 1074, in check_version
    raise Errors.NoBrokersAvailable()
kafka.errors.NoBrokersAvailable: NoBrokersAvailable
[2025-07-25T19:18:24.447+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=sample_ETL, task_id=load_order_and_delivery_data, run_id=manual__2025-07-25T19:17:57.744600+00:00, execution_date=20250725T191757, start_date=20250725T191818, end_date=20250725T191824
[2025-07-25T19:18:24.500+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 40 for task load_order_and_delivery_data (NoBrokersAvailable; 211)
[2025-07-25T19:18:24.524+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-07-25T19:18:24.599+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-25T19:18:24.628+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
